pkg/git/gogit.go

```diff
@@ -34,15 +35,28 @@ import (
 	"github.com/weaveworks/weave-gitops/pkg/utils"
 )
 
+type GotGit struct{}
+
+func (g *GotGit) PlainCloneContext(ctx context.Context, path string, isBare bool, o *gogit.CloneOptions) (*gogit.Repository, error) {
+	return gogit.PlainCloneContext(ctx, path, isBare, o)
+}
+
+//counterfeiter:generate . GoGitI
+type GoGitI interface {
```

Comment:
Don't add the `I` here. We know its an interface :wink: .

Reply:
@josecordaz I would move this interface to a separate package like "wrapper" and drop the `I`.

Reply:
@josecordaz I would also put the GotGit{} struct in the wrapper package and create a constructor that is public which returns the interface.

Reply:
@jpellizzari actually I didn't know how to name it, as I already have a struct named GoGit, any suggestion ?